java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -323,9 +324,13 @@ protected static CommandManager buildCommandManager(ChannelHandler channelHandle
     @SuppressWarnings("unchecked")
     protected <T> T handleRedisResponse(
             Class<T> classType, EnumSet<ResponseFlags> flags, Response response) throws RedisException {
+        boolean encodingUtf8 = flags.contains(ResponseFlags.ENCODING_UTF8);
         boolean isNullable = flags.contains(ResponseFlags.IS_NULLABLE);
         Object value =
-                new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer).apply(response);
+                encodingUtf8
+                        ? new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer).apply(response)
+                        : new BaseCommandResponseResolver(RedisValueResolver::valueFromPointerBinary)
+                                .apply(response);
```

Comment:
As we discussed, we will always return `byte[]` for `Value::BulkString`, so you don't need second resolver and related changes

Reply:
This is an intermediate code that will be removed later (once everything is using GlideString). Without this change, we will have basically all the test broken. With this change,  I can split the work among multiple engineers.

Reply:
Ack. Please add this to the PR description - future steps and current considerations. It will be easier to review and you can avoid such comments.